### PR TITLE
V2.0 aid

### PR DESCRIPTION
**Pull request by ricco386**

### CODE REVIEW

**Comment by ZoltanOnody** [comment]
tests are failing please fix them

File: konfera/models/ticket_type.py [hunk 1]
```diff
@@ -86,9 +87,23 @@ def status(self):
     def issued_tickets(self):
         return self.ticket_set.exclude(status=Ticket.CANCELLED).count()
 
+    def calculate_available_tickets(self, numeric_only=False, negative_usage=False):
+        if self.usage == -1:
+            if numeric_only:
+                left = 32767
```
Comment (ZoltanOnody) [inline konfera/models/ticket_type.py]: where did this number came from?

Comment (ZoltanOnody) [reply konfera/models/ticket_type.py]: why don't we use `float('inf')` ?

File: konfera/models/ticket_type.py [hunk 2]
```diff
@@ -86,9 +87,23 @@ def status(self):
     def issued_tickets(self):
         return self.ticket_set.exclude(status=Ticket.CANCELLED).count()
 
+    def calculate_available_tickets(self, numeric_only=False, negative_usage=False):
```
Comment (ZoltanOnody) [inline konfera/models/ticket_type.py]: I don't like the structor of this method, my personal preference would be to add there more `return` this way I have to look on the whole method to see that the value of `left` doesn't change anymore.

**Review by ZoltanOnody — CHANGES_REQUESTED** [review]
otherwise LGTM

File: konfera/models/ticket_type.py (hunk 2 above)
Comment (ricco386) [reply konfera/models/ticket_type.py]: Well, I think it really depends, in theory, we should minimize the number of returns in each routine. It's harder to understand a routine if, reading it at the bottom, you're unaware of the possibility that it returned somewhere above.

Use a return when it enhances readability. In certain routines, once you know the answer, you want to return it to the calling routine immediately. If the routine is defined in such a way that it doesn't require any cleanup, not returning immediately means that you have to write more code.

I have tried to rewrite the function to use less if's however we do not return 0 if the value is negative somehow, but I guess it make sence if available tickets is -7 than you can no longer sell ;)

**Comment by ricco386** [comment]
Unitests are (were always) OK, codecov is failing as it doesnt have version compare agains (not sure why).

**Review by ZoltanOnody — APPROVED** [review]

**Pull request merged**